frontend/src/lib/k8s/cluster.ts

```diff
@@ -36,6 +36,29 @@ export interface Cluster {
   [propName: string]: any;
 }
 
+export class BaseKubeObject {
+  static getBaseObject(): KubeObjectInterface {
+    return {
+      apiVersion: 'v1',
+      kind: '',
```

Comment:
Once this is moved to the cluster.ts's KubeObject, you should fill this will the appropriate values from the endpoint.

Reply:
apiVersion is now filled from the endpoint and kind is filled by the resource class since the KubeObject class's `className` is not valid by default